src/box/iproto.cc

```diff
@@ -1505,6 +1611,26 @@ tx_inject_delay(void)
 	});
 }
 
+static inline void
+tx_start_request(struct cmsg *m)
+{
+	struct iproto_msg *msg = container_of(m, struct iproto_msg, base);
+	struct session *session = msg->connection->session;
+	session->requests_count++;
```

Comment:
You increment session request count in tx thread, but really request starts in iproto thread! See `iproto_enqueue_batch`!
So some requests can be lose.
Server:
```lua
box.cfg {
        iproto_threads = 1,
        listen = "/tmp/1"
}
box.schema.user.grant('guest','execute,write,read','universe')
box.ctl.set_on_shutdown_timeout(20)
require('console').start()
```
Client:
```
fiber = require('fiber')

os.execute("rm -rf *.snap *.xlog")

ccc = 1000
net_box = require('net.box')
table = {}
for i = 1, ccc do
    graceful_con = net_box.connect("/tmp/1")
    graceful_con.space._session_settings:update('graceful_shutdown',$
    table[i] = graceful_con
end

for i = 1, ccc do
    fiber.create(function() for i = 1, 10000 do
            table[i]:ping()
        end
    end)
end
require('console').start()
```

Reply:
Added to the test